src/main.tf

```diff
@@ -33,15 +33,16 @@ resource "azurerm_resource_group" "main" {
 }
 
 resource "azurerm_cosmosdb_account" "main" {
-  name                                  = var.md_metadata.name_prefix
-  location                              = azurerm_resource_group.main.location
-  resource_group_name                   = azurerm_resource_group.main.name
-  offer_type                            = "Standard"
-  kind                                  = "MongoDB"
-  enable_automatic_failover             = var.geo_redundancy.automatic_failover
-  enable_multiple_write_locations       = var.geo_redundancy.multi_region_writes
-  mongo_server_version                  = var.database.mongo_server_version
-  public_network_access_enabled         = false
+  name                            = var.md_metadata.name_prefix
+  location                        = azurerm_resource_group.main.location
+  resource_group_name             = azurerm_resource_group.main.name
+  offer_type                      = "Standard"
+  kind                            = "MongoDB"
+  enable_automatic_failover       = var.geo_redundancy.automatic_failover
+  enable_multiple_write_locations = var.geo_redundancy.multi_region_writes
+  mongo_server_version            = var.database.mongo_server_version
+  # With public_network_access_enable = true, plus virtual_network_filter_enabled = true, the CosmosDB account will be accessible from the VNet and not the internet.
+  public_network_access_enabled         = true
```

Comment:
This seems sketch. Is this just REALLY poorly named? Any time "public access" is true for a database it seems like something is wrong. Is there any documentation we can reference saying this is the proper way to do it?

Reply:
This is what happens when I set the value to false:
![image](https://github.com/massdriver-cloud/azure-cosmosdb-mongo/assets/8037512/c4264a91-d095-4a30-9c5e-ad090c3acb92)
And then it refuses the connection:
![image](https://github.com/massdriver-cloud/azure-cosmosdb-mongo/assets/8037512/4f272e2b-2bf1-48e3-b5fc-c1eec9762671)

Reply:
This is with the private endpoint properly configured:
![image](https://github.com/massdriver-cloud/azure-cosmosdb-mongo/assets/8037512/eefb7ab8-266b-4417-8fb5-ffb9f1ed3e52)
This is the documentation that is found on this page: https://learn.microsoft.com/en-us/azure/cosmos-db/how-to-configure-firewall
This is not how it's supposed to work. I'm supposed to be able to set `Disabled` aka `public_network_access_enabled=false` and set up a private endpoint and it works. But, it doesn't. And Microsoft couldn't figure out why yesterday. 

I have a new support ticket to get a documentable answer from them. I won't merge this PR until I have that answer.

Reply:
Not sure if this is relevant:
https://learn.microsoft.com/en-us/azure/cosmos-db/how-to-configure-private-endpoints?tabs=arm-bicep#blocking-public-network-access-during-account-creation

Reply:
Definitely relevant. Followed their guide on how to set this up to a T. Still doesn't work. According to starter network support, a private endpoint isn't the only setup needed. We'd also need to create and configure a load balancer that points traffic to the private endpoint, and set up DNS so there's a publicly accessible hostname. And a route table so that traffic is routed properly. 

This setup meets requirements of network isolation and major security compliances. We have a second customer now waiting on this network issue to be resolved.